nvml-wrapper/src/device.rs

```diff
@@ -289,6 +289,49 @@ impl<'nvml> Device<'nvml> {
         }
     }
 
+    /**
+    Gets the NUMA nodes physically close to the GPU.
+
+    Main goal is to facilitate memory placement optimisations for multi CPU/GPU settings.
+    Node (set) size needs to be something like `<Number of nodes> / (std::mem::size_of::<c_ulong>() / 8) + 1`
+
+    # Errors
+
+    * `Uninitialized`, if the library has not been successfully initialized
+    * `InvalidArg`, if this `Device` is invalid
+    * `NotSupported`, if this `Device` does not support this query
+    * `GpuLost`, if this `Device` has fallen off the bus or is otherwise inaccessible
+    * `Unknown`, on any unexpected error
+    */
+    // Checked against local
+    // Tested
+    #[cfg(target_os = "linux")]
+    #[doc(alias = "nvmlDeviceGetMemoryAffinity")]
+    pub fn memory_affinity(
+        &self,
+        size: usize,
+        scope: nvmlAffinityScope_t,
+    ) -> Result<Vec<c_ulong>, NvmlError> {
+        let sym = nvml_sym(self.nvml.lib.nvmlDeviceGetMemoryAffinity.as_ref())?;
+
+        unsafe {
+            if size == 0 {
+                return Err(NvmlError::InsufficientSize(Some(1)));
+            }
+
+            let mut affinities: Vec<c_ulong> = vec![mem::zeroed(); size];
```

Comment:
Using `0` here is clearer, and less unsafe.

```suggestion
            let mut affinities: Vec<c_ulong> = vec![0; size];
```